### PR TITLE
Resolve intents even if no MVCC versions need GCing.

### LINKED ISSUE #2497: Resolve intents even when there are no MVCC versions to GC.
as explained in #2495:
`gcQueue` does two kinds of GC: old MVCC versions and transactions/intents. However, if there are no MVCC versions to GC it bails out early and doesn't attempt to clean up any intents: https://github.com/cockroachdb/cockroach/blob/bd8b2de68d85ae6af495826f17f8daae48d2f06d/storage/gc_queue.go#L218

I think we just need to rearrange that code in `gc_queue.go`: move the "Process push transactions in parallel" block up above "Set start and end keys". 

This would remove the need for the initial Put calls in `storage/gc_queue_test.go: TestGCQueueIntentResolution`

### PR DESCRIPTION
**Pull request by mberhault**
Fixes #2497

### CODE REVIEW

File: storage/gc_queue.go [hunk 1]
```diff
@@ -239,10 +232,23 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
 		}
 	}
 
+	done := true
 	if len(intents) > 0 {
+		done = false
```
Comment (bdarnell) [inline storage/gc_queue.go]: I don't think this needs to set `done` (and therefore `done` doesn't need to exist this early). If intents exist, we resolve them, and that's it.

File: storage/gc_queue.go [hunk 2]
```diff
@@ -239,10 +232,23 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
...
+		gcArgs.Key = gcArgs.Keys[0].Key
+		gcArgs.EndKey = gcArgs.Keys[len(gcArgs.Keys)-1].Key.Next()
+	}
+
+	if done {
+		return nil
+	}
+
 	// Send GC request through range.
```
Comment (bdarnell) [inline storage/gc_queue.go]: We can only send the GC request if `gcArgs.Keys` is not empty. I think the old logic was right to return when `gcArgs.Keys` was empty; the only thing we need to change is move `repl.resolveIntents` up before that return.

Comment (mberhault) [inline storage/gc_queue.go]: the recipient of the GCRequest does two things:
* `MVCCGarbageCollect` which iterates over all keys in the `GCRequest`
* `MVCCPutProto` of the `GCRequest.GCMeta`. This one has `OldestIntentNanos` set if we have `len(intents) > 0`.

Without the `GCRequest`, the `GCMeta` keeps its old value.
At the very least, this is what the test looks for after the call to `gcqueue.process`. Whether that's the right thing to check or not, I can't tell, but it seems to be.

Comment (mberhault) [inline storage/gc_queue.go]: more specifically, when `len(keys) == 0`, nothing happens, `MVCCGarbageCollect` is really just a big range over those keys, with nothing before or after.

**Comment by bdarnell** [comment]
Got it. LGTM

**Pull request merged**